data_prep/project_context/context_retriever.py

```diff
@@ -24,7 +26,8 @@ def __init__(self, project_name: str, function_signature: str):
     self._project_name = project_name
     self._function_signature = function_signature
     self._download_from_path = f'{self.AST_BASE_PATH}/{self._project_name}/*'
-    self._ast_path = f'{self.DOWNLOAD_TO_PATH}/{self._project_name}'
+    self._uuid = uuid.uuid4()
+    self._ast_path = f'{self.DOWNLOAD_TO_PATH}/{self._project_name}{self._uuid}'
 
   def _get_function_name(self, target_function_signature: str) -> str:
     """Retrieve the function name from the target function signature."""
```

Comment:
nit: s/Retrieve/Retrieves/ for consistency.

Reply:
done